strawberryfields/backends/tfbackend/ops.py

```diff
@@ -445,22 +460,22 @@ def squeezed_vacuum(r, theta, D, pure=True, batched=False):
     return squeezed
 
 
-def displaced_squeezed(alpha, r, phi, D, pure=True, batched=False, eps=1e-12):
+def displaced_squeezed(r_d, phi_d, r_s, phi_s, D, pure=True, batched=False, eps=1e-12):
     """creates a single mode input displaced squeezed state"""
-    alpha = tf.cast(alpha, def_type)
-    r = (
-        tf.cast(r, def_type) + eps
+    alpha = tf.cast(r_d, def_type) * tf.exp(1j * tf.cast(phi_d, def_type))
+    r_s = (
+        tf.cast(r_s, def_type) + eps
     )  # to prevent nans if r==0, we add an epsilon (default is miniscule)
-    phi = tf.cast(phi, def_type)
+    phi_s = tf.cast(phi_s, def_type)
 
-    phase = tf.exp(1j * phi)
-    sinh = tf.sinh(r)
-    cosh = tf.cosh(r)
-    tanh = tf.tanh(r)
+    phase = tf.exp(1j * phi_s)
+    sinh = tf.sinh(r_s)
+    cosh = tf.cosh(r_s)
+    tanh = tf.tanh(r_s)
 
     # create Hermite polynomials
     gamma = alpha * cosh + tf.math.conj(alpha) * phase * sinh
-    hermite_arg = gamma / tf.sqrt(phase * tf.sinh(2 * r))
+    hermite_arg = gamma / tf.sqrt(phase * tf.sinh(2 * r_s))
 
```

Comment:
this could potentially be written as recursion instead of taking powers and factorials, although that can be done in subsequent PR.

Reply:
I see you're enjoying recursion now 😄

Reply:
Learnt my lessons from the Lord of Recursion ;)